### DISCUSSION TITLE
Async API calls

### DISCUSSION THREAD

**Discussion by foundrytom** [OpenAssetIO/OpenAssetIO#35] [Ideas and change proposals]
Generally, the hosts we’ve investigated resolve assets synchronously (the thread waits for the call to return).

Presently, OAIO is a synchronous system. However, a given manager implementation can be asynchronous, for example, when making calls to a backend system.

With the above considerations, a few questions arise:

Q: Is there a need for asynchronous method calls in OAIO? If yes - any operations in particular?
Q: Is it likely that asynchronous asset resolution (or more general use of the API) would become more common in hosts?

**Comment by pshinners-blz** [comment]
Our studio's system is fully synchronous, I don't see that changing soon.

If there are hosts that could take advantage of async, I could imagine a host firing up some worker threads, OAIO objects, and a worker queue. That could be done entirely outside OAIO. It could even be provided as a utility with OAIO libraries without the managers needing to know about it.

To me, supporting batch requests handles a majority of cases where efficiency is needed, while remaining vastly simpler.
